### PR TITLE
[TextFields] Remove usage of MDCOutlinedTextFieldColorThemer

### PR DESCRIPTION
**Pull request by andrewoverton**
This goal of this PR is to remove usage of MDCTextFieldColorThemer in preparation for its deprecation and deletion. The snapshot tests that were using this themer no longer have color theming. There is a TODO in each of these tests saying to use a theming extension if/when one that will work for them is written.

Related to #9080

### CODE REVIEW

File: components/TextFields/src/Theming/MDCTextInputControllerOutlined+MaterialTheming.m [hunk 1]
```diff
@@ -28,7 +33,32 @@ - (void)applyThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme {
 }
 
 - (void)applyColorThemeWithScheme:(nonnull id<MDCColorScheming>)colorScheme {
-  [MDCOutlinedTextFieldColorThemer applySemanticColorScheme:colorScheme toTextInputController:self];
+  UIColor *onSurfaceOpacity =
```
Comment (romoore) [inline components/TextFields/src/Theming/MDCTextInputControllerOutlined+MaterialTheming.m]: Please update the Theming extension in a separate PR and then delete the themer in its own PR.

**Review by romoore — DISMISSED** [review]

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m [hunk 2]
```diff
@@ -44,9 +44,6 @@ - (void)setUp {
   MDCSemanticColorScheme *colorScheme =
       [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
```
Comment (romoore) [inline components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:43]: Shouldn't this be replaced with a call to the theming extension?

Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:43]: MDCTextInputControllerOutlinedTextArea doesn't have a theming extension. Only MDCTextInputControllerOutlined and MDCTextInputControllerFilled have theming extensions. The approach I'm taking in this PR is sort of similar to https://github.com/material-components/material-components-ios/pull/9196

File: components/TextFields/src/Theming/MDCTextInputControllerOutlined+MaterialTheming.m (hunk 1 above)
Comment (andrewoverton) [reply components/TextFields/src/Theming/MDCTextInputControllerOutlined+MaterialTheming.m]: I was definitely planning on deleting the themer in its own PR, after yet another PR for deprecating it. I can break out the theming extension change.

Comment (andrewoverton) [reply components/TextFields/src/Theming/MDCTextInputControllerOutlined+MaterialTheming.m]: This PR changes the theming extension: https://github.com/material-components/material-components-ios/pull/9223

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m (hunk 2 above)
Comment (romoore) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:43]: Thanks. Would you please leave a TODO at each location with a link to an Issue that requests the MDCTextInputControllerOutlinedTextArea theming extension?

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m [hunk 3]
```diff
@@ -41,11 +41,6 @@ - (void)setUp {
       [[MDCTextInputControllerOutlinedTextArea alloc] initWithTextInput:self.textField];
   self.textFieldController.characterCountMax = 50;
   self.textFieldController.characterCountViewMode = UITextFieldViewModeAlways;
-  MDCSemanticColorScheme *colorScheme =
```
Comment (romoore) [inline components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:44]: On line 15, can you remove the ColorThemer imports? I would also expect the BUILD/Podspec file to be updated with the removed dependencies for the examples.

**Review by romoore — APPROVED** [review]
LGTM pending deps clean-up and TODOs.

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m (hunk 2 above)
Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:43]: I think the TODOs left for #9196 are sufficient. Where the typography (and now color theming) code used to be it now says `// TODO: Theme the MDCTextInputController with a theming extension (to be added in #9199)`

Comment (romoore) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:43]: Ah, sorry I didn't see those. Looks good!

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m (hunk 3 above)
Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:44]: Removed imports and BUILD deps 👍

**Pull request merged**